mapstring.cpp

```diff
@@ -1896,6 +1896,86 @@ char* msConvertWideStringToUTF8 (const wchar_t* string, const char* encoding)
 #endif
 }
 
+wchar_t* msConvertWideStringFromUTF8 (const char* string, const char* encoding)
+{
+#ifdef USE_ICONV
+  wchar_t* output = NULL;
+  char* errormessage = NULL;
+  iconv_t cd = NULL;
+  size_t nStr;
+  size_t nInSize;
+  size_t nOutSize;
+  size_t iconv_status = -1;
+  size_t nBufferSize;
+
+  const char* pszUTF8 = NULL;
+  wchar_t* pwszWide = NULL;
+
+  if (string != NULL) {
+    nStr = strlen (string);
+    nBufferSize = ((nStr * 6) + 1);
+    output = (wchar_t*) msSmallMalloc (nBufferSize);
+
+    if (nStr == 0) {
+      /* return an empty 8 byte string */
+      output[0] = '\0';
+      return output;
+    }
+
+    cd = iconv_open(encoding, "UTF-8");
+
+    nOutSize = nBufferSize;
+    if ((iconv_t)-1 != cd) {
+      nInSize = sizeof (char)*nStr;
+      pszUTF8 = string;
+      pwszWide = output;
+      iconv_status = iconv(cd, &pszUTF8, &nInSize, (char **)&pwszWide, &nOutSize);
```

Comment:
For the linux builds the warnings are treated as error, so we need to handle the cast from const char* to char* in the second argument